code/game/machinery/cloning.dm

```diff
@@ -284,32 +303,46 @@
 
 	H.suiciding = 0
 	attempting = 0
+
+	if(reagents && reagents.total_volume)
+		reagents.reaction(occupant, 2, 1000)
+		reagents.trans_to(occupant, 1000)
+
 	return 1
 
 //Grow clones to maturity then kick them out.  FREELOADERS
 /obj/machinery/clonepod/process()
-	var/show_message = 0
-	for(var/obj/item/weapon/reagent_containers/food/snacks/meat/meat in range(1, src))
-		qdel(meat)
-		biomass += BIOMASS_MEAT_AMOUNT
-		show_message = 1
-	if(show_message)
-		visible_message("[src] sucks in and processes the nearby biomass.")
-
 	if(stat & NOPOWER) //Autoeject if power is lost
 		if(occupant)
 			locked = 0
 			go_out()
 			connected_message("Clone Ejected: Loss of power.")
+		return
 
-	else if((occupant) && (occupant.loc == src))
-		if((occupant.stat == DEAD) || (occupant.suiciding))  //Autoeject corpses and suiciding dudes.
+	if(occupant && occupant.loc == src)
+		if(occupant.stat == DEAD || occupant.suiciding) // Autoeject corpses and suiciding dudes.
 			locked = 0
 			announce_radio_message("The cloning of <b>[occupant]</b> has been aborted due to unrecoverable tissue failure.")
 			go_out()
 			connected_message("Clone Rejected: Deceased.")
+			return
+
+		if(failed_tick_counter >= MAX_FAILED_CLONE_TICKS) // you been in there too long, get out
+			locked = 0
+			announce_radio_message("The cloning of <b>[occupant]</b> has been aborted due to low biomatter.")
+			go_out()
+			connected_message("Clone Ejected: Low Biomatter.")
+			return
 
-		else if(occupant.cloneloss > (100 - heal_level))
+		if(meat_level <= 0)
+			failed_tick_counter++
+			if(failed_tick_counter == (MAX_FAILED_CLONE_TICKS / 2)) // halfway to ejection.
+				connected_message("Low Biomatter: Preparing to Eject Clone")
+				announce_radio_message("Low Biomatter: Preparing to Eject <b>[occupant]</b>")
+				use_power(200)

```

Comment:
it's idle power usage since the actual idle power usage won't ever be called if it returns here